### DISCUSSION TITLE
Git LFS (large file storage)

### DISCUSSION THREAD

**Discussion by MFAshby** [community/community#5240] [Codespaces] [answered]
Hi! 

We use `git lfs` in our project to store some large files as part of our git repo. It looks like `git lfs` isn't installed automatically, and large files are not automatically pulled on codespace creation. 

It would be great if repositories that used `git lfs` would automatically pull large files!

**Comment by tanmayeekamath** [comment, marked as answer]
Hey @MFAshby, thanks for your feedback. While we try to include most components in the[ default Codespaces image](https://aka.ms/ghcs-default-image), we recommend using the devcontainer configuration to customize the Codespaces based on your scenario and project needs :) 
Here's more on customizing your Codespace: https://docs.github.com/en/codespaces/customizing-your-codespace/configuring-codespaces-for-your-project

You can either include installing git lfs as a part of your postcreatecommand section of your devcontainer.json file or include it in the container image itself. Here's more on devcontainer.json file properties: https://code.visualstudio.com/docs/remote/devcontainerjson-reference#_devcontainerjson-properties

Hope this helps!

**Reply by MFAshby** [reply]
Thanks Tanmayee, I'll check out that postcreatecommand :+1: 

I have git lfs extension installed in the container, it just doesn't automatically run on the checkout of the git repository inside the container.

**Reply by MFAshby** [reply]
I'm trying to mark this as the answer but failing :D 

This was the answer anyway, thanks!
